crypto/s2n_hash.c

```diff
@@ -205,26 +211,26 @@ static int s2n_evp_hash_init(struct s2n_hash_state *state, s2n_hash_algorithm al
         r = 1;
         break;
     case S2N_HASH_MD5:
-        r = EVP_DigestInit_ex(state->digest.evp.primary.ctx, EVP_md5(), NULL);
+        r = EVP_DigestInit_ex(state->digest.high_level.evp.ctx, EVP_md5(), NULL);
```

Comment:
I had made these changes but after rebasing my s2n FIPS unit test changes onto this branch, these changes broke the build.

Due to OpenSSL's internal use of EVP_MD_CTX when in FIPS mode, the EVP_Digest_copy API must be used to copy pointers to an EVP_MD_CTX. For this reason, we cannot abstract the accessors within high-level hash functions.

I needed to revert these changes in the latest commit.